pkg/experiment/query_backend/block_reader.go

```diff
@@ -132,3 +155,18 @@ func validateRequest(req *queryv1.InvokeRequest) (*request, error) {
 	}
 	return &r, nil
 }
+
+func filterInvalidDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
```

Comment:
Also a comment would be nice, why we do that:

```suggestion
// While the metastore is expected to already filter datasets of other tenants, we do an additional check to avoid processing blocks or datasets belonging to the wrong tenant.
func filterInvalidDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
```